lib/ts/recipe/multitenancy/allowedDomainsClaim.ts

```diff
@@ -0,0 +1,45 @@
+import { WindowHandlerReference } from "supertokens-website/utils/windowHandler";
+import { SessionClaimValidator, PrimitiveArrayClaim } from "../session";
+
+/**
+ * We include "Class" in the class name, because it makes it easier to import/use the right thing (the instance exported by the recipe) instead of this.
+ * */
+export class AllowedDomainsClaimClass extends PrimitiveArrayClaim<string> {
+    constructor() {
+        super({
+            id: "st-tenant-domains",
+            refresh: async () => {
+                // Can't automatically refresh
+            },
+            defaultMaxAgeInSeconds: Number.MAX_SAFE_INTEGER,
+        });
+
+        // FIXME - maybe it's better if we can add getHost to the interface we are using below, that's in `supertokens-website` npm library
+        const currentDomain = WindowHandlerReference.getReferenceOrThrow()
+            .windowHandler.location.getOrigin()
+            .split("://")[1];
```

Comment:
how will this behave for hybrid apps like:
- capacitor
- electron

Reply:
window.location.origin is unavailable in electron app.